rudder-lang/src/main.rs

```diff
@@ -103,10 +104,18 @@ fn add_file<'a>(
     path: &'a Path,
     filename: &'a str,
 ) -> Result<()> {
-    let content = fs::read_to_string(path)
-        .unwrap_or_else(|_| panic!("Something went wrong reading the file {}", filename));
-    let content_str = source_list.append(content);
-    past.add_file(filename, &content_str)
+    println!("|- {} {}", "Parsing".bright_green(), filename);
+    match fs::read_to_string(path) {
+        Ok(content) => {
+            let content_str = source_list.append(content);
+            past.add_file(filename, &content_str)
+        },
+        Err(e) => {
+            let tk = Token::new(filename, "");
+            println!("{:#?}", tk);
+            Err(err!(Token::new(filename, ""), "{}", e))
```

Comment:
You should be able to use `tk` in the `Err` (you will need to pass `&tk` to the `println` though)

Reply:
It was actually just some debug I forgot, correcting right now